src/systemctl/systemctl.c

```diff
@@ -9089,7 +9089,9 @@ static int logind_schedule_shutdown(void) {
                 return log_warning_errno(r, "Failed to call ScheduleShutdown in logind, proceeding with immediate shutdown: %s", bus_error_message(&error, r));
 
         if (!arg_quiet)
-                log_info("Shutdown scheduled for %s, use 'shutdown -c' to cancel.", format_timestamp(date, sizeof(date), arg_when));
+                log_info("%s scheduled for %s, use 'shutdown -c' to cancel.",
+                         (arg_action == ACTION_REBOOT ? "Reboot" : "Shutdown"),
```

Comment:
This *does* fix the immediate issue reported in #16129, but not the other cases. If you look at the `switch` statement above, it is clear that the message is inappropriate in other cases too, e.g. for kexec it'll still say "Shutdown". So maybe extend the `switch` statement to also specify a human-readable action name for this message:
```C
case ACTION_HALT:
    action = "halt";
    human = "Halt";
    break;
...
case ACTION_KEXEC:
   action = "kexec";
   human = "Reboot via kexec";
   ...
```

Also, coding style: the parentheses in the added line are not needed. But that'll be resolved if the code is reworked as suggested anyway.

Reply:
@keszybz, I think I have resolved the raised points in 5ad42057f65d23d2b51b0aba8cdd38b1daae1196. If you want I can squash my two commits.